### PR TITLE
Generic type bounds for FS types

### PR DESCRIPTION
**Pull request by kustosz**
Changes the type bounds on `prove` and `verify` to not hardcode keccak for challenges. This is required to allow these methods to work with algebraic hashes.

### CODE REVIEW

File: src/skyscraper.rs
Comment (WizardOfMenlo) [inline src/skyscraper.rs:1]: I'm guessing this is not meant for this PR?

**Review by WizardOfMenlo — APPROVED** [review]
LGTM! Let's just remove the skyscraper.rs file

File: src/skyscraper.rs
Comment (kustosz) [reply src/skyscraper.rs:1]: yes this is accidental LMAO

Comment (kustosz) [reply src/skyscraper.rs:1]: Removed :)

**Pull request merged**